src/governance/governor/Governor.sol

```diff
@@ -75,45 +75,36 @@ contract Governor is IGovernor, VersionedContract, UUPS, Ownable, EIP712, Propos
     /// @notice Initializes a DAO's governor
     /// @param _treasury The DAO's treasury address
     /// @param _token The DAO's governance token address
-    /// @param _vetoer The address eligible to veto proposals
-    /// @param _votingDelay The voting delay
-    /// @param _votingPeriod The voting period
-    /// @param _proposalThresholdBps The proposal threshold basis points
-    /// @param _quorumThresholdBps The quorum threshold basis points
-    function initialize(
-        address _treasury,
-        address _token,
-        address _vetoer,
-        uint256 _votingDelay,
-        uint256 _votingPeriod,
-        uint256 _proposalThresholdBps,
-        uint256 _quorumThresholdBps
-    ) external initializer {
+    /// @param _data The encoded governor parameters
+    function initialize(address _treasury, address _token, bytes calldata _data) external initializer {
         // Ensure the caller is the contract manager
         if (msg.sender != address(manager)) revert ONLY_MANAGER();
 
         // Ensure non-zero addresses are provided
         if (_treasury == address(0)) revert ADDRESS_ZERO();
         if (_token == address(0)) revert ADDRESS_ZERO();
 
+        GovParams memory params = abi.decode(_data, (GovParams));
+
         // If a vetoer is specified, store its address
-        if (_vetoer != address(0)) settings.vetoer = _vetoer;
+        if (params.vetoer != address(0)) settings.vetoer = params.vetoer;
 
         // Ensure the specified governance settings are valid
-        if (_proposalThresholdBps < MIN_PROPOSAL_THRESHOLD_BPS || _proposalThresholdBps > MAX_PROPOSAL_THRESHOLD_BPS)
+        if (params.proposalThresholdBps < MIN_PROPOSAL_THRESHOLD_BPS || params.proposalThresholdBps > MAX_PROPOSAL_THRESHOLD_BPS)
             revert INVALID_PROPOSAL_THRESHOLD_BPS();
-        if (_quorumThresholdBps < MIN_QUORUM_THRESHOLD_BPS || _quorumThresholdBps > MAX_QUORUM_THRESHOLD_BPS) revert INVALID_QUORUM_THRESHOLD_BPS();
-        if (_proposalThresholdBps >= _quorumThresholdBps) revert INVALID_PROPOSAL_THRESHOLD_BPS();
-        if (_votingDelay < MIN_VOTING_DELAY || _votingDelay > MAX_VOTING_DELAY) revert INVALID_VOTING_DELAY();
-        if (_votingPeriod < MIN_VOTING_PERIOD || _votingPeriod > MAX_VOTING_PERIOD) revert INVALID_VOTING_PERIOD();
+        if (params.quorumThresholdBps < MIN_QUORUM_THRESHOLD_BPS || params.quorumThresholdBps > MAX_QUORUM_THRESHOLD_BPS)
+            revert INVALID_QUORUM_THRESHOLD_BPS();
```

Comment:
nit: typically i like the syntax of forcing `{` and `}` in these situations to prevent updates messing up conditionals.